src/commands/qt_args/q_arg_float.py

```diff
@@ -0,0 +1,50 @@
+from PySide6.QtWidgets import QAbstractSpinBox, QDoubleSpinBox
+from src.commands.qt_args.q_arg_value import QArgValue
+
+
+class QArgFloat(QArgValue):
+    ALLOWED_TYPES = (type(None), float, int)
+
+    def __init__(
+        self,
+        name: str,
+        default_value: float = 0,
+        min_value: float | None = None,
+        max_value: float | None = None,
+        unit: str = "",
+        description: str = None,
+    ) -> None:
+        """Create a new QArgFloat
+
+        Args:
+            name (str): argument name
+            default_value (float, optional): arg default value. Defaults to 0.
+            min_value (float | None, optional): arg min value. Defaults to None.
+            max_value (float | None, optional): arg max value. Defaults to None.
+            unit (str, optional): arg unit. Defaults to "".
+            description (str, optional): arg description. Defaults to None.
+        """
+        super().__init__(name, default_value, min_value, max_value, unit, description)
+
+    def _check_type(self, value: int | float | None) -> bool:
```

Comment:
to avoid defining this method in multiple places I guess you could just add this in the QArgBasic class alongside ALLOWED_TYPES which you can overload (if this is how python works idk)

Reply:
I moved code from this method to QArgValue along with ALLOWED_TYPES, and overloaded the variable with the appropriate types.

Reply:
I moved the body of this method to QArgValue along with ALLOWED_TYPES, then in qargfloat and qargint override ALLOWED_TYPES with appropriate values.